wazuh/wazuh-qa
# Add authorization context JWT invalidation system tests

Hello team,

this PR adds new system tests for the JWT invalidation module. These ones log in using the authorzation context.

Closes #840 .

# Test checks

- Check that an existing token is revoked when updating the password from the user that requested it.
- Check that an existing token is revoked when changing roles related to the user that requested it.
- Check that an existing token is revoked when changing policies related to the user that requested it.
- Check that an existing token is revoked when changing rules related to the user that requested it.
- Check that every token is revoked when making a call to `PUT /security/user/revoke`.
- Check that both master and worker nodes retrieve token information when one of them disconnects from the cluster.
- Check that an existing token is revoked when changing the RBAC mode.

# Test results
```
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.8.2, pytest-5.4.2, py-1.8.1, pluggy-0.13.1
rootdir: /home/vicferpoy/Desktop/Git/wazuh-qa/tests/system
plugins: asyncio-0.12.0, testinfra-5.0.0, tavern-1.0.0
collected 20 items                                                                                                                                                                          

test_jwt_invalidation/test_change_rbac_mode.py ....                                                                                                                                   [ 20%]
test_jwt_invalidation/test_change_security_resources.py ....                                                                                                                          [ 40%]
test_jwt_invalidation/test_disconnected_nodes.py .                                                                                                                                    [ 45%]
test_jwt_invalidation/test_revoke_endpoint.py ......                                                                                                                                  [ 75%]
test_jwt_invalidation/test_update_password.py .....                                                                                                                                   [100%]

========================================================================= 20 passed in 1106.70s (0:18:26) =========================================================================

```

Best regards.
